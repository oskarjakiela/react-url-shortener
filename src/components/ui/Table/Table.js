import classnames from 'classnames';
import PropTypes from 'prop-types';
import { addIndex, map } from 'ramda';
import React from 'react';

import './Table.css';


const mapIndexed = addIndex(map);
const displayName = 'Shorty-Table';

const Table = ({
  className,
  headers,
  rows,
  ...props
}) => (
  <table className={classnames(displayName, className)} {...props}>
    <thead>
      <tr>
        {map(header => (
          <th key={header}>{header}</th>
        ), headers)}
      </tr>
    </thead>

    <tbody>
      {mapIndexed((values, row) => (
        <tr key={row}>
          {mapIndexed((value, cell) => (
            <td key={`${row}-${cell}`}>{value}</td>
          ), values)}
        </tr>
      ), rows)}
    </tbody>
  </table>
);

Table.displayName = displayName;

Table.propTypes = {
  headers: PropTypes.arrayOf(PropTypes.string).isRequired,
  rows: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.node)).isRequired,
};

export default Table;
