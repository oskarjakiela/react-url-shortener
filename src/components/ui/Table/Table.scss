.Shorty-Table {
  td,
  th {
    padding: .5rem;
    text-align: center;

    &:first-child {
      padding-left: 0;
    }

    &:last-child {
      padding-right: 0;
    }
  }

  th {
    font-size: .875rem;
    font-weight: 400;
    text-transform: uppercase;
    color: #aaa;
    white-space: nowrap;
  }
}
