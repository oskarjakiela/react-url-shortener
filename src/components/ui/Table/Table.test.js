import React from 'react';
import renderer from 'react-test-renderer';

import Table from './Table';


it('<Table /> renders correctly', () => {
  const tree = renderer
    .create(<Table
      headers={['Foo']}
      rows={[
        [<span>Bar</span>]
      ]}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
