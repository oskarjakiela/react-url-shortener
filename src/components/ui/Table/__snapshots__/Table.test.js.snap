// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> renders correctly 1`] = `
<table
  className="Shorty-Table"
>
  <thead>
    <tr>
      <th>
        Foo
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <span>
          Bar
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
